agent/provider/src/execution/registry.rs

```diff
@@ -146,16 +146,18 @@ pub struct ExeUnitsRegistry {
     descriptors: HashMap<String, ExeUnitDesc>,
 }
 
-impl ExeUnitsRegistry {
-    pub fn new() -> ExeUnitsRegistry {
-        ExeUnitsRegistry {
-            descriptors: HashMap::new(),
+impl Default for ExeUnitsRegistry {
+    fn default() -> Self {
+        Self {
+            descriptors: Default::default(),
         }
     }
+}
```

Comment:
I've removed `new` functions in some places but they can be left and added only `Default`. I am not sure if we want to have both or just `default()`